llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -935,6 +937,7 @@ void AMDGPUTargetMachine::registerPassBuilderCallbacks(PassBuilder &PB) {
             PM.addPass(InternalizePass(mustPreserveGV));
             PM.addPass(GlobalDCEPass());
           }
+          PM.addPass(AMDGPUSplitKernelArgumentsPass());
```

Comment:
Also, I think we might want to think carefully where to add the pass. Will it potentially benefit other passes? It looks like for now it is added to where the attributor pass runs, which is actually almost the end of the pipeline. Is it expected?